news/_posts/2022-07-01-introducing-horizontal-migration.md

```diff
@@ -0,0 +1,74 @@
+
+Following that logic, we could say that the first step to migrating a controller is "transforming" it into a Symfony controller. Formally speaking, this requires minimal changes to the original code. It might not be pretty, but it provides all the benefits of a full-fledged Symfony controller (routing, kernel events, debug features, translator, container, etc) – for a low, low price.
+
+Migrating a page using this "shortcut" should take fewer modifications than doing it the traditional way. This means less developer effort (making it easier for contributors to participate), less effort to review and, hopefully, less effort to verify. If we did this to all remaining controllers, we could have 100% of the Back office running on Symfony controllers much sooner than if we performed all the refactoring steps to each of them as we do now. This would allow us to phase out Dispatcher, Link, AdminController, and other legacy components much faster.
+
+Now, rest assured, this wouldn't mean abandoning Twig, Grid, Symfony forms, and CQRS – quite the opposite. Once we had all controllers running on Symfony, we could start the second step: removing Smarty in favor of Twig. If we managed to make HelperList and HelperForm work using Twig, we could easily phase out Smarty. This would simply require refactoring the helpers (in which case we wouldn't need to change any of the controllers), or creating new components that follow the same signature as the old helpers (which would require minimal changes to the controllers). Again, this could be applied everywhere, allowing us to phase out Smarty, the legacy Back office theme, and the legacy layout that envelops Symfony Twig pages.
+
+After that, we could go on to phase out HelperLists in favor of Grids, then HelperForms in favor of Symfony forms and CQRS. And then, we would get a complete migration of the Back office.
+
+We call this approach “horizontal migration”, because it focuses on replacing one layer of legacy components with its modern equivalent across all the pages before moving on to the next layer. In opposition, we call our current approach "vertical migration" because it focuses on migrating all the layers of a page at once. The end result is the same: full migration to Symfony. It’s the path to get there that changes.
+
+## What's next?
+
+We think the horizontal approach provides two clear benefits compared to our current approach:
+- A quicker reduction of the system's overall complexity, because legacy components can be sunset as soon as they are no longer needed, instead of having to wait until the end of the migration to do so.
```

Comment:
How about this? "wait until the migration ends" makes me feel that the migration would be done by itself 😄 
```suggestion
- - A quicker reduction of the system's overall complexity, because legacy components can be sunset as soon as they are no longer needed, instead of having to wait until the whole migration has been completed.
```